src/registrar/decorators.py

```diff
@@ -397,3 +395,60 @@ def _is_staff_managing_domain(request, **kwargs):
     # the user is permissioned,
     # and it is in a valid status
     return True
+
+
+def _has_domain_requests_view_all(user, request, domain_request_pk):
+    """
+    Determines if the user has view-all permission for domain requests.
+    This permission allows users to view domain request details without editing.
+    Handles both portfolio and non-portfolio domain requests.
+    """
+    logger.debug(f"Checking DOMAIN_REQUESTS_VIEW_ALL for user={user} pk={domain_request_pk}")
+
+    portfolio = request.session.get("portfolio")
+    # Portfolio-based access
+    if user.is_org_user(request) and portfolio:
+        has_perm = user.has_view_all_domain_requests_portfolio_permission(portfolio)
+        exists = _domain_request_exists_under_portfolio(portfolio, domain_request_pk)
+        logger.debug(f"Portfolio check → perm={has_perm}, exists={exists}")
+        return has_perm and exists
+
+    # No PK means “free” view
+    if not domain_request_pk:
+        logger.warning(
+            "_has_domain_requests_view_all: missing domain_request_pk; " "assuming view-all is allowed in this context."
+        )
+        return True
+
+    # Check non-portfolio permissions
+    try:
+        dr = DomainRequest.objects.get(pk=domain_request_pk)
```

Comment:
code convention: can we change this variable name for consistency?
```suggestion
        request = DomainRequest.objects.get(pk=domain_request_pk)
```